### ISSUE TITLE
Importing from stdlib modules using the deprecated name gives an error

### ISSUE THREAD

**Issue by c42f** [JuliaLang/julia#24832]
I'd expect to be able to import with a depwarn, but get an error instead:

```julia
julia> import Base.Test: record
WARNING: Base.Test is deprecated, run `using Test` instead.
  likely near no file:0
ERROR: invalid import path: "Test" does not name a module
```

**Comment by ararslan** [comment]
Related to (possibly duplicate of) #24488

**Comment by vchuravy** [comment]
definitly a dup and I might have a fix for this.

**Comment by KristofferC** [comment]
Fixing this would be incredibly helpful in reducing breakage.

**Comment by vchuravy** [comment]
This is now properly fixed even for exported functions and submodules after #25692
